client/src/controllers/ZoneController.ts

```diff
@@ -0,0 +1,72 @@
+import { Controller } from '@hotwired/stimulus';
+import { debounce } from '../utils/debounce';
+
+enum ZoneMode {
+  Active = 'active',
```

Comment:
@ayaan-qadri  I also pulled these strings out to TypeScript enums. This enables easier expansion if needed in the future.